src/Generator/Generators/CLI/CLITypeReferences.cs

```diff
@@ -191,7 +191,8 @@ public override bool VisitClassDecl(Class @class)
             if (@class.IsIncomplete && @class.CompleteDeclaration != null)
                 @class = (Class) @class.CompleteDeclaration;
 
-            var keywords = @class.IsValueType? "value struct" : "ref class";
+            var keywords = (@class.IsValueType && !(@class.HasBaseClass && @class.BaseClass.IsRefType))
```

Comment:
This doesn't look right to me. We should be changing the value of `@class.IsValueType` in a pass instead if the class is not supposed to be a value type.